hibernate-core/src/main/java/org/hibernate/internal/util/StringHelper.java

```diff
@@ -708,12 +708,15 @@ public static String unquote(String name) {
 	 *
 	 * @return True if quoted, false otherwise
 	 */
-	public static boolean isQuoted(String name, Dialect dialect) {
-		return name != null && name.length() != 0
-				&& ( ( name.charAt( 0 ) == '`' && name.charAt( name.length() - 1 ) == '`' )
-				|| ( name.charAt( 0 ) == '"' && name.charAt( name.length() - 1 ) == '"' )
-				|| ( name.charAt( 0 ) == dialect.openQuote()
-				&& name.charAt( name.length() - 1 ) == dialect.closeQuote() ) );
+	public static boolean isQuoted(final String name, final Dialect dialect) {
+		if ( name == null || name.isEmpty() ) {
+			return false;
+		}
+		final char first = name.charAt( 0 );
+		final char last = name.charAt( name.length() - 1 );
+
+		return ( ( first == last ) &&
+				( first == '`' || first == '"' || first == dialect.closeQuote() ) );
```

Comment:
The above is not quite right, because first and last may not be the same.  For example, ```SQLServerDialect.openQuote()``` returns ```'['```, and closeQuote() returns ```']'```.

For example,

Reply:
It should be:

```return first == dialect.openQuote() && last == dialect.closeQuote();```

Reply:
@Sanne, or should the fix avoid calling the dialect methods, if possible?

Reply:
Good catch. There is a different isQuoted method in which a similar optimisations was valid, but not here.